src/base/private/profile_p.cpp

```diff
@@ -109,6 +109,8 @@ Private::CustomProfile::CustomProfile(const QString &rootPath, const QString &co
 
 QString Private::CustomProfile::baseDirectory() const
 {
+    if (!QDir().mkpath(m_rootDirectory.absolutePath()))
+        qFatal("Could not create required directory '%s'", qUtf8Printable(m_rootDirectory.absolutePath()));
     return m_rootDirectory.canonicalPath();
```

Comment:
If directory doesn't exist `m_rootDirectory.canonicalPath()` returns empty string.

Reply:
Sorry, what does m_rootDirectory represent? E.g. if profile dir is `E:\qBittorrent\profile` then what does `CustomProfile::baseDirectory()` return?

Reply:
If profile dir is `E:\qBittorrent\profile` then `profile->baseDirectory()` before https://github.com/qbittorrent/qBittorrent/blob/e906478a16937236bdf583c3d36ef13175eaba20/src/base/profile.cpp#L55-L57 returns empty string if qBittorrent was launched for the first time (`profile` dir is empty), and `E:/qBittorrent/profile/qBittorrent` on all consecutive runs.

If we ensure directory exists then it will be always returning `E:/qBittorrent/profile/qBittorrent`.
I think it's ok to unsure here that `m_rootDirectory` exists because qBittorrent will not work if it can't be created anyway.